js/chennai.js

```diff
@@ -1,349 +1,246 @@
+  $('#map').toggleClass('loading');
+  var url = DATASETS_BASE + 'features/' + features[0].properties.id + '?access_token=' + DATASETS_ACCESS_TOKEN;
+  var index = addedRoads.indexOf(features[0].properties.id);
+  $.ajax({
+    method: 'DELETE',
+    url: url,
+    contentType: 'application/json',
+    success: function () {
+      $('#map').toggleClass('loading');
+      data['features'].splice(index, 1);
+      addedRoads.splice(index, 1);
+      addedFeatures.splice(index, 1);
+      SELECTED_ROADS_SOURCE.setData(data);
+      updateFeatureCount(data);
+    },
+    error: function () {
+      $('#map').toggleClass('loading');
+    }
+  });
+}
 
-        //Dump Data
-        window.dump = JSON.stringify(data);
+function addRoad (data, addedRoads, addedFeatures, SELECTED_ROADS_SOURCE, features) {
+  $('#map').toggleClass('loading');
+  var tempObj = {
+    type: 'Feature',
+    geometry: features[0].geometry,
+    properties: features[0].properties,
+  };
+  tempObj.properties['is_flooded'] = true;
+  tempObj.id = md5(JSON.stringify(tempObj));
+
+  var url = DATASETS_BASE + 'features/' + tempObj.id + '?access_token=' + DATASETS_ACCESS_TOKEN;
+  $.ajax({
+    method: 'PUT',
+    url: url,
+    data: JSON.stringify(tempObj),
+    dataType: 'json',
+    contentType: 'application/json',
+    success: function (response) {
+      $('#map').toggleClass('loading');
+      tempObj.id = response.id;
+      tempObj.properties.id = response.id;
+      addedFeatures.push(tempObj);
+      data.features.push(tempObj);
+      addedRoads.push(features[0].properties.osm_id);
+      SELECTED_ROADS_SOURCE.setData(data);
+      updateFeatureCount(data);
+    },
+    error: function () {
+      $('#map').toggleClass('loading');
+    }
+  });
+}
 
-        for (var i = 0; i < data.features.length; i++) {
-            addedRoads.push(data.features[i].properties.id);
-            addedFeatures.push(data.features[i]);
-        }
+function addSourcesAndLayers() {
+  $('#feature-count').toggleClass('loading');
+  SELECTED_ROADS_SOURCE = new mapboxgl.GeoJSONSource({});
+  map.addSource('selected-roads', SELECTED_ROADS_SOURCE);
+  map.addLayer({
+    'id': 'selected-roads',
+    'type': 'line',
+    'source': 'selected-roads',
+    'interactive': true,
+    'paint': {
+      'line-color': 'rgba(255,5,230,1)',
+      'line-width': 3,
+      'line-opacity': 0.6
+    }
+  }, 'road-waterlogged');
 
+  map.addSource('terrain-data', {
+    type: 'vector',
+    url: 'mapbox://mapbox.mapbox-terrain-v2'
+  });
+  map.addLayer({
+    'id': 'terrain-data',
+    'type': 'line',
+    'source': 'terrain-data',
+    'source-layer': 'contour',
+    'layout': {
+      'line-join': 'round',
+      'line-cap': 'round'
+    },
+    'paint': {
+      'line-color': '#ff69b4',
+      'line-opacity': '0.3',
+      'line-width': 1
+    }
+  });
+  $('#feature-count').toggleClass('loading');
+}
+
+function selectRoad(data) {
```

Comment:
Minor nitpick: I wonder if this function should be named differently since it doesn't actually "select road", but rather sets up the event handler that allows a road to be selected?